src/intrinsics/ecma262/Object.js

```diff
@@ -118,8 +130,30 @@ export default function(realm: Realm): NativeFunctionValue {
       }
     }
 
+    if (to_must_be_abstract) {
+      let types = new TypesDomain(FunctionValue);
+      let values = new ValuesDomain();
+      invariant(realm.generator);
+      to = realm.generator.derive(types, values, [target, ...sources], _args => {
+        return t.callExpression(
+          t.memberExpression(t.identifier("Object"), t.identifier("assign")),
+          ((_args: any): Array<any>)
+        );
+      });
+      if (target instanceof ObjectValue) {
+        // we have to make the target partial and simple, because it was
+        // an object before and needs to be treated the same to prevent failures
+        target.makePartial();
+        target.makeSimple();
```

Comment:
I still don't understand this. There's nothing as far as I can see that guarantees `target` is simple. What gives us the knowledge to claim it is simple now?

Reply:
We're in a pure scope now, so we can make those assertions.

Reply:
Hmm I think I see. You're saying that if we're in pure scope, we promised not to mutate any objects created before we entered it. So even if we have an object with a getter that has a side effect, presumably we promise not to run it?

I don't think I understand if this is safe. This will need @sebmarkbage or @hermanventer to take a look.